File: api_java/src/main/java/todo/api_java/model/TodoEntryDTO.java
package todo.api_java.model;


import todo.api_java.enums.TodoStatus;
import todo.api_java.enums.TodoType;

import java.sql.Date;

public class TodoEntryDTO {
    private String id;
    private String text;
    private TodoStatus status;
    private TodoType type;
    private Date creationDate;
    private Date lastChecked;
}

File: api/src/models/todoEntryDTO.ts
export enum TodoStatus {
  OPEN = "OPEN",
  DONE = "DONE",
  DELETED = "DELETED"
};

export enum TodoType {
  SINGLE = "SINGLE",
  PERIODIC = "PERIODIC"
};

export enum TimeInSeconds {
  DAY = 60 * 60 * 24,
  WEEK = DAY * 7
}

export interface TodoEntryDTO {
  id: string,
  text: string,
  description?: string,
  status: TodoStatus, 
  creationDate: string,
  type: TodoType,
  lastChecked?: string,
  periodSeconds?: number,
}

export type TodoCreateDTO = Omit<TodoEntryDTO, "id">;